service/controller/key/key.go

```diff
@@ -21,10 +24,36 @@ const (
 	ServiceTypeManaged = "managed"
 )
 
-func ReleaseVersion(customResource releasev1alpha1.Release) string {
-	return customResource.Spec.Version
+// ReleaseAppCRName returns the name of the release App CR for the given release cycle.
+func ReleaseAppCRName(releaseCycleCR releasev1alpha1.ReleaseCycle) string {
+	return releasePrefix(releaseCycleCR.GetName())
+}
+
+// ReleasePrefix adds release- prefix to name.
+func ReleasePrefix(name string) string {
+	return fmt.Sprintf("release-%s", name)
+}
+
+// ReleaseVersion returns the version of the given release.
+func ReleaseVersion(releaseCR releasev1alpha1.Release) string {
+	return releaseCR.Spec.Version
+}
+
+// SplitReleaseName splits a release name into provider and version.
+// It returns provider, version, and error, in this order.
+//
+// It expects name to be in the following format <provider>.<version>
+// e.g. aws.v6.0.1
+func SplitReleaseName(name string) (string, string, error) {
```

Comment:
We need unit tests for this. Thinking further about what this method does makes me wonder why we should derive any meta information from the name of anything. The better design would be to use the name to lookup whatever resource we want to fetch and take the relevant information from there. Provider and version should be tracked in the `Release` CR.

Reply:
> We need unit tests for this.

I agree on this.

> The better design would be to use the name to lookup whatever resource we want to fetch and take the relevant information from there. Provider and version should be tracked in the `Release` CR.

There is no `Release` CR installed yet at the time the `ReleaseCycle` CR is created, since we use the latter to create the former. Hence we cannot fetch the information from the `Release` CR.

We also decided not to store the release name and version as separate fields in the `ReleaseCycle` CR as this open the door to store even more information there like the url of the catalog, kubeconfig and so on. Which would make `ReleaseCycle` CR be a pale copy of `App` CR.

We should be able to rely on the name here to provide us with the relevant information we need. If the name is wrong, it means that something is broken and has to be fixed.

Reply:
Understood. I think that smells but I do not have a better idea now. Let's have unit tests though.

Reply:
added unit test. PTAL